apps/backend/src/action/logic/pass.ts

```diff
@@ -1,21 +1,28 @@
-import { ResultAsync, errAsync } from "neverthrow";
+import { ResultAsync, errAsync, okAsync } from "neverthrow";
 import { EventSource } from "~/models/event";
 import { PassCommand } from "~/action/command";
 import { TurnHasBeenPassedEvent } from "~/action/event";
 import { always, error } from "~/utils";
 
+import { Event as ActionEvent } from "~/action/event";
+import { P, match } from "ts-pattern";
+import { pipe } from "fp-ts/function";
+
+const RepositoryWriteError = error("RepositoryWriteError");
 export type RepositoryWriteError = ReturnType<typeof RepositoryWriteError>;
-export type PassError = RepositoryWriteError;
+
+const PassFailedError = error("PassFailedError");
+export type PassFailedError = ReturnType<typeof PassFailedError>;
+
+export type PassError = RepositoryWriteError | PassFailedError;
 
 export interface Pass {
   (
     source: EventSource<TurnHasBeenPassedEvent>,
     command: PassCommand
-  ): ResultAsync<TurnHasBeenPassedEvent[], Error /* PassError */>;
+  ): ResultAsync<TurnHasBeenPassedEvent[], PassError>;
```

Comment:
could you help me update this by 
```ts
export type Pass = (
  source: EventSource<Event>,
  command: PassCommand
) => ResultAsync<TurnHasBeenPassedEvent, PassError>;
```

Reply:
The `Event` type could be found in `~/action/event`

Reply:
ok ~  !